code/game/turfs/simulated/water.dm

```diff
@@ -108,11 +115,109 @@
 		return species.can_breathe_water()
 	return ..()
 
-// Use this to have things react to having water applied to them.
-/atom/movable/proc/water_act(amount)
-	return
-
-/mob/living/water_act(amount)
-	adjust_fire_stacks(-amount * 5)
-	for(var/atom/movable/AM in contents)
-		AM.water_act(amount)
\ No newline at end of file
+/turf/simulated/floor/beach/water/proc/wash_floor(var/turf/T)
+	T.clean(src)
+
+// Taken from shower
+/turf/simulated/floor/beach/water/proc/wash(atom/movable/O as obj|mob)
```

Comment:
This proc looks duplicated from elsewhere; this probably should be made a helper proc instead of copied.

Reply:
It is a helper proc, the functionality was borrowed from shower, as comment states

Reply:
Apparently I can't read.